server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -848,22 +844,23 @@ private boolean isEmptyOrTrivialWildcard(List<String> expressions) {
 
     static final class DateMathExpressionResolver implements ExpressionResolver {
 
+        private static final DateFormatter DEFAULT_DATE_FORMATTER = DateFormatters.forPattern("uuuu.MM.dd");
         private static final String EXPRESSION_LEFT_BOUND = "<";
         private static final String EXPRESSION_RIGHT_BOUND = ">";
         private static final char LEFT_BOUND = '{';
         private static final char RIGHT_BOUND = '}';
         private static final char ESCAPE_CHAR = '\\';
         private static final char TIME_ZONE_BOUND = '|';
 
-        private final DateTimeZone defaultTimeZone;
+        private final ZoneId defaultTimeZone;
         private final String defaultDateFormatterPattern;
-        private final DateTimeFormatter defaultDateFormatter;
+        private final DateFormatter defaultDateFormatter;
 
         DateMathExpressionResolver(Settings settings) {
             String defaultTimeZoneId = settings.get("date_math_expression_resolver.default_time_zone", "UTC");
-            this.defaultTimeZone = DateTimeZone.forID(defaultTimeZoneId);
-            defaultDateFormatterPattern = settings.get("date_math_expression_resolver.default_date_format", "YYYY.MM.dd");
-            this.defaultDateFormatter = DateTimeFormat.forPattern(defaultDateFormatterPattern);
+            this.defaultTimeZone = ZoneId.of(defaultTimeZoneId);
+            defaultDateFormatterPattern = settings.get("date_math_expression_resolver.default_date_format", "uuuu.MM.dd");
```

Comment:
Since this is in 6.8, shouldn't this default format be prefixed with '8' since we still use joda by default here?

Reply:
yes, should be with `8`. will fix